code/datums/votes/map_vote.dm

```diff
@@ -4,12 +4,16 @@
 	count_method = VOTE_COUNT_METHOD_SINGLE
 	winner_method = VOTE_WINNER_METHOD_NONE
 	display_statistics = FALSE
+	/// Amount of viable players during out last validity check
+	var/cached_pop = -1
 
 /datum/vote/map_vote/New()
 	. = ..()
+	cached_pop = length(GLOB.clients)
 	default_choices = SSmap_vote.get_valid_map_vote_choices()
 
 /datum/vote/map_vote/create_vote()
+	cached_pop = length(GLOB.clients)
 	default_choices = SSmap_vote.get_valid_map_vote_choices()
```

Comment:
why do we not just have a wrapper proc to update choices if pop has changed, isn't this potentially wasteful due to the call in `can_be_initiated()`?